bigtable/admin.go

```diff
@@ -1033,6 +1045,41 @@ func (iac *InstanceAdminClient) InstanceInfo(ctx context.Context, instanceID str
 	}, nil
 }
 
+// AutoscalingConfig contains autoscaling configuration for a cluster.
+// For details, see https://cloud.google.com/bigtable/docs/autoscaling.
+type AutoscalingConfig struct {
+	// MinNodes sets the minumum number of nodes in a cluster.
+	MinNodes int
+	// MaxNodes sets the maximum number of nodes in a cluster.
+	MaxNodes int
+	// CPUTargetPercent sets the CPU utilization target for your cluster's
+	// workload.
+	CPUTargetPercent int
+}
+
+func (a AutoscalingConfig) isZero() bool {
+	return a.MinNodes == 0 && a.MaxNodes == 0 && a.CPUTargetPercent == 0
+}
+
+func (a AutoscalingConfig) proto() *btapb.Cluster_ClusterConfig_ {
```

Comment:
It seems a bit odd to have an `AutoscalingConfig` convert to a new `Cluster_ClusterConfig` rather than say modifying an existing one. Is it possible that other cluster config fields will be added in the future?

Reply:
That's a good point. I'll update this function to just return the corresponding type. If future fields are added, I think this method will be open for extension.

Reply:
This method makes sense to me (but maybe I'm thinking about this wrong). I think about proto message objects as being like value-only structs--they're a way to pass structured data around.

I'm not sure what we gain by having this method modify an existing `Cluster_ClusterConfig_` proto message.

Reply:
Tightened this up to have it return the corresponding proto type.